pkg/controller/daemon/controller.go

```diff
@@ -432,7 +503,16 @@ func (dsc *DaemonSetsController) manage(ds *extensions.DaemonSet) {
 	for i := 0; i < createDiff; i++ {
 		go func(ix int) {
 			defer createWait.Done()
-			if err := dsc.podControl.CreatePodsOnNode(nodesNeedingDaemonPods[ix], ds.Namespace, ds.Spec.Template, ds); err != nil {
+
+			// Make a copy
+			template := *ds.Spec.Template
+			// Add podTemplateHash to pod template labels.
+			template.ObjectMeta.Labels = labelsutil.CloneAndAddLabel(
+				ds.Spec.Template.ObjectMeta.Labels,
+				ds.Spec.UniqueLabelKey,
+				podutil.GetPodTemplateSpecHash(template))
+
+			if err := dsc.podControl.CreatePodsOnNode(nodesNeedingDaemonPods[ix], ds.Namespace, &template, ds); err != nil {
```

Comment:
I'd also like to copy annotations from the DaemonSet to the pods it creates. The main reason is so that history-related annotations (change-source, note, etc.) are propagated.